### ISSUE TITLE
Internal Server Error after docker-compose down docker-compose up -d

### ISSUE THREAD

**Issue by deeztek** [nextcloud/docker#1457]
Fresh install of Nextcloud 20 on docker-compose. When I start a fresh clean container for NC 20, it starts up fine with no issues. If I do a docker-compose down and then docker-compose up -d, I get the following error:

```
Internal Server Error

The server encountered an internal error and was unable to complete your request.
Please contact the server administrator if this error reappears multiple times, please include the technical details below in your report.
More details can be found in the server log.
```

there is nothing of value in the nextcloud.log file that I can see. Nothing of value in the container logs. If I start the container using docker-compose up in order to see logs, I'm not seeing any errors. This is really weird because the container start up with no problems the first time but never works again if I do a docker-compose down. If I delete the volumes and do a brand new container it works again. This is obviously less than ideal. I would appreciate some help.

Thanks

**Comment by ryujaehun** [comment]
I'm having the same issue

**Comment by deeztek** [comment]
So for me it seemed to be the **image: mariadb** in the example docker-compose. What was happening the first time I would bring up the stack, the database would get created, nextcloud would install and then everything would be fine until I brought the stack down and the supposedly persistent **db** volume would get removed. So, when I would bring the stack back up, nextcloud was already set as **'installed' => true,** in the **config.php** file and it would try to connect to the non-existent database. The way I figured that out is by doing a docker-compose up without the "-d" part so I could see the output when the container was being brought up and I saw a bunch of missing table errors.  So it seems that the native mariadb docker image is ignoring persistent volumes for some reason.

So long story short, I used the **image: linuxserver/mariadb** image and it worked fine. Here's an example:

```
  db:
    image: linuxserver/mariadb
    container_name: nextcloud_db
    logging:
      options:
        max-size: 10m
    environment:
      - TZ=${TIMEZONE}
      - PUID=${PUID}
      - PGID=${GUID}
    env_file:
      - db.env
    volumes:
      - db:/config
    networks:
      - proxy
    restart: unless-stopped

volumes:
  db:
    driver: local
    driver_opts:
      type: none
      o: bind
      device: /mnt/data/db
```

Hopefully this helps.

**Comment by mrkz** [comment]
I'm being affected by this issue as well.

**Comment by NajibNour** [comment]
Hey hey, Seems like I'm facing this issue as well. I have tried updating libseccomp with solution 3 but that had no effect. I've also tried creating the folders before even launching the compose file from clean and the issue is still present.

Docker-compose down (Breaks nextcloud)
docker stop NXdb NXapp (does not break nextcloud I can then start it without issues.)
Using linuxserver/mariadb (Ive used this on multiple other containers without issues.)

when launching docker-compose up after running docker-compose down and watching the log it seems to be repopulating the db tables once more. Not too sure how I can resolve that ...

**Comment by NajibNour** [comment]
Well, I'm silly. After 4h of trying to find the issue, I finally got it. I'm not sure how I didn't catch this early. When using the linuxserver/mariadb image map it to /config and not /var/lib/mysql.

**Comment by joshtrichards** [comment]
Hi @deeztek - Glad you were able to find a solution for your situation. I'm not sure what the deal was with your db persistent volume - I've never encountered any issues with it and it's impossible to reproduce without your compose file. Further, it's outside the scope of this repository - which is intended for reporting bugs in the Nextcloud community Docker image. :-) If anyone has any follow-up, I suggest they post at the [Nextcloud Help Forum](https;//help.nextcloud.com/).

If this Issue is no longer relevant, can you close it out here to avoid confusion? Thanks!
